linter.py

```diff
@@ -24,18 +37,19 @@ class PHP(Linter):
         'selector': 'source.php, text.html.basic'
     }
     regex = (
-        r'^(?:Parse|Fatal) (?P<error>error):(\s*(?P<type>parse|syntax) error,?)?\s*'
-        r'(?P<message>(?:unexpected \'(?P<near>[^\']+)\')?.*) (?:in - )?on line (?P<line>\d+)'
+        r'^(?P<error>Parse|Fatal) error:\s*'
+        r'(?P<message>((?:parse|syntax) error,?)?\s*(?:unexpected \'(?P<near>[^\']+)\')?.*) '
+        r'(?:in - )?on line (?P<line>\d+)'
     )
     error_stream = util.STREAM_STDOUT
 
     def split_match(self, match):
         """Return the components of the error."""
-        match, line, col, error, warning, message, near = super().split_match(match)
+        result = super().split_match(match)
+
+        match, line, col, error, warning, message, near = result
 
```

Comment:
IIRC you don't have to unpack `result`.  You can just `result["message"] = _filter_message(result.message)` (mutate 😨!) or so.   What `super().split_match(match) returns is usually named `error` (instead of `result`).  (Basically `split_match` transforms a regex `Match` to a dict like thing.)